syntaxes/textproto.tmLanguage.json

```diff
@@ -81,6 +86,24 @@
                 }
             ]
         },
+        "heredoc": {
+            "begin": "(<<)\\s*([_a-zA-Z]+)\\s*$",
+            "beginCaptures": {
+                "1": {
+                    "name": "keyword.operator.heredoc.textproto"
+                },
+                "2": {
+                    "name": "keyword.control.heredoc-token.textproto"
+                }
+            },
+            "end": "^\\s*(\\2)\\s*$",
+            "endCaptures": {
+                "1": {
+                    "name": "keyword.control.heredoc-token.textproto"
+                }
+            },
+            "name": "string.unquoted.heredoc.no-indent.textproto"
+        },
```

Comment:
<<EOF
Had no idea you could capture and use the capture in the end pattern
EOF